File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        #sort on the basis of ending time
        points = sorted(points, key=lambda x: x[1])
        currPointer = points[0][1]
        arrows = 1

        for i in range(len(points)):
            if points[i][0] > currPointer: #if start of i > last pointer pointing at
                currPointer = points[i][1]
                arrows += 1
        return arrows

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.go
func findMinArrowShots(points [][]int) int {
    slices.SortFunc(points, func(a, b []int) int {
        return a[1] - b[1]
    })

    cur := points[0][1]
    cnt := 1
    for i := 1; i < len(points); i++ {
        if points[i][0] > cur {
            cnt += 1
            cur = points[i][1]
        }
        
    }

    return cnt
}